src/tiling/tile_state.rs

```diff
@@ -136,11 +136,15 @@ impl<'a, T: Pixel> TileStateMut<'a, T> {
       height % MI_SIZE == 0,
       "Tile width must be a multiple of MI_SIZE"
     );
+
+    let sb_rounded_width = width.align_power_of_two(sb_size_log2);
```

Comment:
Yeap,  how dumb I am I have not done this way from the start and just use them to change two lines below!